skyportal/models.py

```diff
@@ -103,6 +103,35 @@ def is_owned_by(self, user_or_token):
     raise NotImplementedError(f"{type(self).__name__} object has no owner")
 
 
+def is_modifiable_by(self, user_or_token):
+    """Return a boolean indicating whether an object point can be modified or
+    deleted by a given user or token.
+
+    Parameters
+    ----------
+    user_or_token: `baselayer.app.models.User` or `baselayer.app.models.Token`
+       The User or Token to check.
+
+    Returns
+    -------
+    owned: bool
+       Whether the Object can be modified by the User or Token.
+    """
+
+    if not hasattr(self, 'owner'):
+        raise TypeError(
+            f'Object {self} does not have an `owner` attribute, '
+            f'and thus does not expose the interface that is needed '
+            f'to check for modification or deletion privileges.'
+        )
+
+    is_admin = "System admin" in [acl.id for acl in user_or_token.acls]
+    owns_spectrum = self.owner is (
+        user_or_token if isinstance(user_or_token, User) else user_or_token.created_by
+    )
```

Comment:
If we always pass in `self.associated_user_object` from handlers, this can be simplified
```suggestion
    owns_spectrum = self.owner is user_or_token
```
w/ `user_or_token` -> `user`

Reply:
done